### ISSUE TITLE
Plant mutants suffering vitamin deficiency

### ISSUE THREAD

**Issue by Saint-of-Grey** [CleverRaven/Cataclysm-DDA#29241]
**Describe the bug**  
The description for plant mutations suggests that they could survive off soil and sunlight alone. Doing exactly this on my character for over a month has left me with some massive vitamin deficiencies. A peek at the debug menu tells me I'm around -4k for most of my vitamins.

**To Reproduce**  
1. Be a plant mutant with any of the root mutations.
2. Only use that to satisfy your food and drink needs for a long period of time.
3. Suffer from massive vitamin deficiency.

**Expected behavior**  
I thought rooting in dirt would cover my food crisis, but alas it did not. The amount of food I have to eat to address my deficiencies seems to completely offset the benefit of rooting.

**Versions and configuration(please complete the following information):**
 - OS: Windows 8.1
 - Game Version: 0.D-1319-gdca9d26
 - Graphics version: Tiles
 - Mods loaded: Dark Days Ahead, Garden Pots, Makeshift Items Mod, More Survival Tools, Mutant NPCs, More Locations, Urban Development, Fuji's More Buildings, Boats, Folding Parts Pack, Vehicle Additions Pack, Tanks and Other Vehicles, Filthy Clothing, Manual Bionic Installation, Safe Autodoc

**Comment by golankra123** [comment]
Do you eat varied amount of fruit?

Could you list them?
Which vitamins you miss?

**Comment by Saint-of-Grey** [comment]
A handful of oranges, apples, lemons, and bananas. They rotted before I found them, but as a saprophage I didn't really care. Before I got roots I survived off some junk food, but was pretty much on the brink of starvation unit I mutated roots. That was my only food source for over a month.

I'm currently bouncing back between beginning stage vitamin A deficiency and scurvy, offsetting it with a handful of multivitamins I found on a zombie. My vitamin B12 deficiency has progressed to the "No Healing" point.

**Comment by golankra123** [comment]
Im not sure if roots should be a easy win for vitamins. 

How ever pretty much like our ansenstors try to forge food from bushes , and hunt animals.


Vitamin A can be got by meat or fish easily. 
Same with b12.

Although it should be checked if there any plants that give you those vitamins (cause there should be)

**Comment by Saint-of-Grey** [comment]
So there are no vitamins in the soil at all?

I feel like if that is intentional, the mutation description should at least mention it.

**Comment by Photoloss** [comment]
> Im not sure if roots should be a easy win for vitamins.

Definitely not, plants do generally rely on soil for **all** their nutrients but not all plants can simply draw everything they need without symbiotic microbes. We see practices such as [green manure](https://en.wikipedia.org/wiki/Green_manure) for demanding crops and fruit trees need fertiliser to keep providing bountiful harvests so there's no way a highly active plant-animal hybrid could satisfy all their needs just by plonking down a few roots for half a day.

Roots or other post-threshold plant mutations certainly should provide some basic vitamins commonly found in plants. Carnivores [get adjusted vitamin rates](https://github.com/CleverRaven/Cataclysm-DDA/blob/815153c27e7cccdfcb25484f2814b2f70c16619f/data/json/mutations.json#L4737) to imitate animal diets so at least the tech exists. Of the currently implemented ones vitA and vitC might be waived entirely as they are mainly gained from plants and vitB demand could be drastically reduced. Looking up dietary advice for vegans the main nutrients plants cannot readily provide (to *normal* humans) are iron, calcium, vitamin D and vitamin B12. Another aspect which is not implemented but worth noting for the future is protein.

Unless someone can come up with an acceptable way of replacing their function (mainly bones and haemoglobin i.e. red blood) I don't see how a plant mutant could get away with just rooting without becoming sessile and vegetative in the process. All that walking around and keeping your brain going takes a lot of effort beyond just shovelling in the sugar.

**Comment by KorGgenT** [comment]
what RDA vitamins would one serving of soil contain?

**Comment by Photoloss** [comment]
> what RDA vitamins would one serving of soil contain?

Impossible to quantify since the roots grow at a supernatural rate, we can't even determine the "portion size". Most vitamins and other organic compounds are actually *synthesised* by the plants rather than absorbed directly which makes it way too complicated. 

> alternatively: would being a plant ( a particular mutation, or threshold ) reduce your need for any particular vitamins? if so, how much?

[Beta-Carotene](https://en.wikipedia.org/wiki/%CE%92-carotene) is a pure hydrocarbon so you're only limited by calories and catalytic enzymes on that one. I'm not a plant biochemist but nothing really stands out in the [Carotenoid synthesis](https://en.wikipedia.org/wiki/Carotenoid#Biosynthesis) which we cannot handwave if desired. [Vitamin C](https://en.wikipedia.org/wiki/Vitamin_C) has a lot of hydroxy groups but no trace elements either.

According to [this](https://en.wikipedia.org/wiki/Vitamin_C#Plant_pathways) anything involving bark, leaves or flowers *could* provide the full pathway if desired for gameplay and threshold or photosynthesis should basically disable vitC needs. Beta-Carotene (vitA) also taps into photosynthesis pathways. B vitamins are an entire group and we definitely won't get B12 for free by going green but [B1 can be produced by plants](https://en.wikipedia.org/wiki/Vitamin_B1#Biosynthesis). 

Essentially for every "vitamin" except some animal-specific amino acids (protein) you *can* find a specific group of plants producing plenty of them. Here's [iron](https://www.healthline.com/nutrition/iron-rich-plant-foods) and [calcium](https://begoodorganics.com/top-75-sources-of-plant-based-calcium/) for your multivitamin plantation.

Since mutations aren't "intelligent design" on a cellular level but the Blob *could* be nudging things in the right direction we basically have full freedom between self-sufficiency and single-digit percentage buffs for the above vitamins. There are no *chemical* limitations as they're only needed in trace amounts and can in principle be synthesised from air, water and sugar. *Metabolically* everything is complicated by the fact we are only *part* plant even post-threshold.

From a gameplay perspective I think vitA and vitC can be switched off by the threshold mutation itself to lessen the debuffs on people who take "living off the land" a little too literally like OP. Also matches the stereotypical expectation of not needing to "eat your greens" when you *are* the greens for what that's worth. Leaving in vitB, calcium and iron requirements (potentially lessened a bit) would preserve *some* need for diet awareness and represent the need to keep our "animal side" going. Those can be obtained from meat and would allow a nomadic "carnivorous plant" lifestyle which I find to be an interesting take on the "naturalist".

**Comment by KorGgenT** [comment]
> > what RDA vitamins would one serving of soil contain?
> 
> Impossible to quantify since the roots grow at a supernatural rate, we can't even determine the "portion size". Most vitamins and other organic compounds are actually _synthesised_ by the plants rather than absorbed directly which makes it way too complicated.

ok, so you won't get vitamins from soil? 


i could agree with lessening vitA and vitC by a significant margin, but i'll also say that you're still at least sapient and i would expect you to still have to bolster your intake some amount. The others i would leave as-is, and since vitamin needs are actually mutation and not threshold based, we would have to decide which mutation gives what. and i'm talking numbers here. like, a normal human needs 100% RDA per day, and [this mutation] "provides" 50% RDA of this vitamin

**Comment by Photoloss** [comment]
> ok, so you won't get vitamins from soil?

No, you don't even need the soil.

> i could agree with lessening vitA and vitC by a significant margin, but i'll also say that you're still at least sapient and i would expect you to still have to bolster your intake some amount. The others i would leave as-is, and since vitamin needs are actually mutation and not threshold based, we would have to decide which mutation gives what. and i'm talking numbers here. like, a normal human needs 100% RDA per day, and [this mutation] "provides" 50% RDA of this vitamin

That approach simply does not work here. *Technically* thresholds include an actual mutation so as far as coding goes we should be able to use that. The issue is that it is entirely *possible* even for an otherwise normal human to metabolise all the vitamins with some genetic modification (not a vitamin but we can do [this](https://www.mpimp-golm.mpg.de/2044732/HIV_vaccine_in_algae)). The necessary chemicals and energy already exist within our bodies. Vitamins are only needed in "trace" quantities (less than 1% by mass and volume) so all we need is the "functional implementation" of synthesis into our metabolism. 

For an analogy, we know freshwater is not chemically pure and boiling the stuff is easy, so how much calcium can you extract from one litre? Well if you collect rainwater basically none but downstream from the mountains [possibly 0.2g of carbonate](https://en.wikipedia.org/wiki/Hard_water#Hard/soft_classification). Except we cannot just "look up the water hardness for New England" because we have no macroscopic plant-animal hybrids to reference.

I am pretty certain any photosynthesis trait should cover all vitA and vitC needs. Not from the actual photosynthesis but because all the supporting metabolic processes would leak enough of them into our bloodstream which our plant cells then replenish. But you could still wiggle out of that one by postulating a strict separation between plant and non-plant organs similar to the blood-brain barrier.

**Comment by Saint-of-Grey** [comment]
> Leaving in vitB, calcium and iron requirements (potentially lessened a bit) would preserve some need for diet awareness and represent the need to keep our "animal side" going. Those can be obtained from meat and would allow a nomadic "carnivorous plant" lifestyle which I find to be an interesting take on the "naturalist".

That certainly would be an interesting take on plant-ness, but I feel like that would go better by giving plants some predatory mutations to nudge players in the right direction. Right now, mutation descriptions such as these:

> "Leaves"
> "All the hair on your body has turned to long, grass-like leaves.  Apart from being physically striking, these provide you with a minor amount of nutrition while in sunlight.  Slightly reduces wet effects."

> Chloromorphosis:
> "Every inch of your skin is packed with chlorophyll and you have strong roots.  Sleeping on diggable soil will satisfy any hunger or thirst you might have."

> Saprophage:
> "You prefer to sustain yourself using your roots and direct nutrient extraction/synthesis.  You can still consume 'food', though, if you have to: you merely prefer it aged a little."

are strongly suggesting that I don't need any food at all. Which is why I injected those plant serums in the first place, so I wouldn't starve to death before being able to leave the lab I was stuck in.

**Comment by kevingranade** [comment]
If you're stuck in a lab, leaves probably should not have worked at all.

``minor amount of nutrition`` is clearly intended to not suggest that it can sustain you.

``Sleeping on diggable soil will satisfy any hunger or thirst you might have.`` is definitely misleading, we need to fix that.

Saprophage doesn't seem to hint one way or the other, though perversely it's the one with the greatest potential to be a sole source of sustenance.

**Comment by Saint-of-Grey** [comment]
> **You prefer to sustain yourself using your roots and direct nutrient extraction/synthesis.**

The the point is that these descriptions convey an overall theme of alternate subsistence.

**Comment by kevingranade** [comment]
I prefer to subsist entirely on pizza and cake, that doesn't mean it's a good idea.

**Comment by Fris0uman** [comment]
Would it be possible to add a new mutation that produce the appropriate vitamin at a fixed rate , or have photosynthesis do it ?

**Comment by Photoloss** [comment]
TL;DR: can we rename `Saprophage` to "Plant Metabolism" and make it grant vitA+vitC? Plants aren't Saprophages anyway, those tend to be fungi.

> Would it be possible to add a new mutation that produce the appropriate vitamin at a fixed rate

Yes that would make sense as far as comparison to real-life plants goes. Call it "Plant Metabolism", "Vitamin Biosynthesis" or something like that. The current interaction between Plant mutants and fertiliser should also be looked at in this case because you need to actually have the correct metabolic options to benefit from *plant* fertiliser.

> Saprophage doesn't seem to hint one way or the other, though perversely it's the one with the greatest potential to be a sole source of sustenance.

The *description* of the `Saprophage` trait *implies* it should be doing this already, the "direct nutrient ~extraction/~ synthesis" part. [Saprophagy](https://en.wikipedia.org/wiki/Saprophagy) is something else entirely though and that ability alone does not guarantee vitamin synthesis. Plus few if any plants actually are Saprophages.

As far as game design goes that's obviously up to Kevin&co.

> or have photosynthesis do it ?

Definitely not in the literal sense, photosynthesis produces sugar from water and carbon dioxide. The photosynthesis *traits* could provide it as an additional feature as I already said, and if the above standalone trait is added instead it definitely has to be a requirement for the photosynthesis traits.

> Maybe also add a symbiosis efect that you get when using your froots often enought " [appropriate name of bateria here] got into your roots, and is devlopping a symbiosis with you : in exchange for ome nutrient it will produce [name of the vitamin here]"

This sounds like a very interesting mechanic for the far future, but excessively complex and hard to track compared to any other mutation. We can't even view our vitamin levels without the debug menu let alone their rate of change or *which* bacteria we have taken in. Separate "diseases" or mutations for different types of bacteria would be rather cumbersome.

**Comment by Fris0uman** [comment]
> which bacteria we have taken in.

Sure, I just meant a catch all symbiote that does what we want. Tracking all the symbiosis is already very hard in real life we should not atempt to do it in game.

**Comment by Photoloss** [comment]
> Sure, I just meant a catch all symbiote that does what we want. Tracking all the symbiosis is already very hard in real life we should not atempt to do it in game.

Okay gotcha. Still sounds like a lot of single-use code especially if we don't *lose* the symbiosis again once acquired even after long periods of *not* rooting.

If it doesn't make it in here this definitely sounds like a nice idea for the planned mutation overhaul in #28277, since they need to set up some long-term tracking anyway it should fit pretty well.

**Comment by Fris0uman** [comment]
Ok so  : 
 - First solution : you root you have x% chance to get effect_symbiosis for n_times at the end of the timer you lose the effect  : "Something changed in your roots, you no longer feel the presence of the symbiote"

- Second solution : effect_symbiosis is applied permanently but if you take antibiotic you lose it, if your hunger level is too low for too long you lose it, if you're iradiated too much you lose it, etc...

**Comment by Saint-of-Grey** [comment]
So you have to carefully curate the various symbiotic organisms that plants need for vitamins? ...like a garden?

Throw that in with something they could use to hunt game and I think plant nutrition would be covered.
